### PR TITLE
EC ProX PCB

### PR DESCRIPTION
**Pull request by Cipulot**
Addition of the EC Pro X PCB firmware. Open source replacement PCB kit for the HHKB Pro 2 and Pro 3 keyboards.
## Description

The EC Pro X is a replacement PCB kit for HHKB Professional 2 ANSI/JIS keyboards as well as the newer Classic/Hybrid Series (colloquially called Pro 3).

Therefore 2 versions of the firmware are included in this PR:
- ansi_iso
- jis

The ANSI and JIS cases are physically different and the actual PCBs have different wirings and features, hence the need of 2 versions instead of the use of separate keymaps.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cipulot/ec_prox/jis/readme.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# EC Pro X JIS
+
+![EC Pro X JIS PCB](https://i.imgur.com/yJdPOcb.png)
```
Comment (keyboard-magpie) [inline keyboards/cipulot/ec_prox/jis/readme.md:3]:
```suggestion
![EC Pro X JIS PCB](https://i.imgur.com/yJdPOcbh.png)
```
Resize to more friendly dimensions

File: keyboards/cipulot/ec_prox/ansi_iso/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# EC Pro X ANSI/ISO
+
+![EC Pro X ANSI/ISO PCB](https://i.imgur.com/TYhUrV0.png)
```
Comment (keyboard-magpie) [inline keyboards/cipulot/ec_prox/ansi_iso/readme.md:3]:
```suggestion
![EC Pro X ANSI/ISO PCB](https://i.imgur.com/TYhUrV0h.png)
```
Resize to more friendly dimensions.

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]

File: keyboards/cipulot/ec_prox/jis/readme.md (hunk 1 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_prox/jis/readme.md]: Done 👍

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/cipulot/ec_prox/jis/rules.mk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CUSTOM_MATRIX = lite
+SRC += analog.c matrix.c ec_switch_matrix.c
```
Comment (drashna) [inline keyboards/cipulot/ec_prox/jis/rules.mk:2]:
```suggestion
QUANTUM_LIB_SRC += analog.c 
SRC += matrix.c ec_switch_matrix.c
```

**Comment by Cipulot** [comment]
@drashna I've applied the same suggested change in the `rules.mk` to the other version too, since it was suggested only for the JIS one.

File: keyboards/cipulot/ec_prox/ansi_iso/ec_switch_matrix.c [hunk 4]
```diff
@@ -0,0 +1,183 @@
+/* Copyright 2023 Cipulot
```
Comment (drashna) [inline keyboards/cipulot/ec_prox/ansi_iso/ec_switch_matrix.c:1]: if possible, this should be moved to something like `keyboards/cipulot/lib` or `keyboards/cipulot/common`, and used by the different boards that need it.

Comment (Cipulot) [reply keyboards/cipulot/ec_prox/ansi_iso/ec_switch_matrix.c:1]: Hi there @drashna , that is the plan eventually. As of now the file in question changes depending  on the board revision and project all together. The functions are gonna be made "board agnostic" in a future revision.
As of now tho it is better to leave it in the keyboard specific folder.

**Comment by keyboard-magpie** [comment]
The last merges for cipulot's boards are all hardware verified, merged under the new rules (1 reviewer approval with all conditions met).

**Pull request merged**